### PR TITLE
Create owner_of_script.gs

### PR DESCRIPTION
**Pull request by siddhimahawar**

### CODE REVIEW

File: owner_of_script.gs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      'headers': header,
+      'payload' : JSON.stringify(body),
+      'muteHttpExceptions': false
+    };
+    var response = UrlFetchApp.fetch('https://logging.googleapis.com/v2/entries:list', options);
+    var json = response.getContentText();
+    resultData = JSON.parse(json);
+    pageToken = resultData.nextPageToken;
+  } while(!resultData.entries);
+  return {"email" : resultData.entries[0].protoPayload.request.brand.supportEmail, "name" : resultData.entries[0].protoPayload.request.brand.displayName, "projectId" : projectId};
```
Comment (shikhagupta1212) [inline owner_of_script.gs]: one param per line

File: owner_of_script.gs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      limit = true;
+      break;
+    }
+    pageToken = resultData.nextPageToken;
+  
+  } while(!resultData.entries);
+  if(limit) {
+    return getOriginalNameAndOwnerOfScript(projectId);
+  } else {
+    return {"email" : resultData.entries[0].protoPayload.response.supportEmail, "name" : resultData.entries[0].protoPayload.response.displayName, "projectId" : projectId};
```
Comment (shikhagupta1212) [inline owner_of_script.gs]: same as above

File: owner_of_script.gs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    resultData = JSON.parse(json);
+    if(!resultData.entries && !resultData.nextPageToken) {
+      limit = true;
+      break;
+    }
+    pageToken = resultData.nextPageToken;
+  
+  } while(!resultData.entries);
+  if(limit) {
+    return getOriginalNameAndOwnerOfScript(projectId);
```
Comment (shikhagupta1212) [inline owner_of_script.gs]: what is the difference b/w this method and getOriginalNameAndOwnerOfScript?

File: owner_of_script.gs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+/**
+* Get the owners of all the cloud projects
+* @param {string} projectType is the types of cloud projects to be checked i.e., CUSTOM_PROJECTs which are user created and SYSTEM_PROJECTs which are system generate
+* @return {Object} Array of objects having name, email of owner and projectId of Apps Script
+*/
+function getOwnersOfAllScripts(projectType) {  
+  var ALL_PROJECTs = listAllCloudProjects();
+  var emailOfOwnerOfScripts = [];
+  var i;
+  for(i=0; i<ALL_PROJECTs.length; i++) {
```
Comment (shikhagupta1212) [inline owner_of_script.gs]: i < ALL_PROJECTs.length

File: owner_of_script.gs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    if (ALL_PROJECTs[i].lifecycleState != 'ACTIVE') {
+      continue;
+    }
+    var projectId = JSON.stringify(ALL_PROJECTs[i].projectId,null,2);
+    if(projectId.indexOf("sys") == 1.0) {
```
Comment (shikhagupta1212) [inline owner_of_script.gs]: Why this check when you have project type?

File: owner_of_script.gs (hunk 3 above)
Comment (siddhimahawar) [reply owner_of_script.gs]: This was the previous commit
I have removed this method from the file

File: owner_of_script.gs (hunk 1 above)
Comment (siddhimahawar) [reply owner_of_script.gs]: Updated

File: owner_of_script.gs (hunk 5 above)
Comment (siddhimahawar) [reply owner_of_script.gs]: I checked whether the current project is robot account generated or user created

File: owner_of_script.gs (hunk 4 above)
Comment (siddhimahawar) [reply owner_of_script.gs]: Updated

File: owner_of_script.gs (hunk 2 above)
Comment (siddhimahawar) [reply owner_of_script.gs]: Updated

File: owner_of_script.gs (hunk 5 above)
Comment (shikhagupta1212) [reply owner_of_script.gs]: for that you have projectType right?
SYSTEM_PROJECT are robot created. So no need for this.

Comment (siddhimahawar) [reply owner_of_script.gs]: projectType is given by the user for which logs have to be taken.
SYSTEM_PROJECT, CUSTOM_PROJECT, ALL_PROJECT, SPECIFIC_PROJECT are enum values of projectType

For a cloud project to check what type of project it is I have used string matching

Comment (shikhagupta1212) [inline owner_of_script.gs]: Cool. But this check is not robust enough. I can create a project named sys-shikhaCustom right?
Also can you check
-  owner of project ie appsdev-apps-dev-script-auth@system.gserviceaccount.com. 
- system projects are only being created under Organization root > system-gsuite > apps-script

Info: https://developers.google.com/apps-script/guides/cloud-platform-projects?hl=en

Comment (siddhimahawar) [reply owner_of_script.gs]: Updated

**Review by piy314 — APPROVED** [review]

**Pull request merged**